lib/hanami/application/routes.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+require_relative 'router'
+
+module Hanami
+  class Application
+    # Application routes
+    #
+    # Users are expected to inherit from this class to define their application
+    # routes.
+    #
+    # @example
+    #   # config/routes.rb
+    #   # frozen_string_literal: true
+    #
+    #   require "hanami/application/routes"
+    #
+    #   module MyApp
+    #     class Routes < Hanami::Application::Routes
+    #       slice :main, at: "/" do
+    #         root to: "home.show"
+    #       end
+    #     end
+    #   end
+    #
+    #   See {Hanami::Application::Router} for the syntax allowed at the class
+    #   level.
+    #
+    # @see Hanami::Application::Router
+    # @since 2.0.0
+    class Routes
+      def self.router
+        @router ||= Router.new(
```

Comment:
@waiting-for-dev `||=` isn't thread-safe, is this code executed within a context that guarantees thread-safety?

This configuration heavily depends on global state of `Hanami.application`. How do we ensure that all these components are ready when we're instantiating the router? Shouldn't this be the role of the caller instead?

Reply:
Not a concern in the new implementation